### PR TITLE
feat: okxwallet

### PR DESCRIPTION
**Pull request by okxwallet**
Add okx wallet to rainbow

### CODE REVIEW

File: examples/with-next-custom-button/pages/_app.tsx [hunk 1]
```diff
@@ -42,6 +43,7 @@ const connectors = connectorsForWallets([
     wallets: [
       argentWallet({ chains }),
       trustWallet({ chains }),
+      okxWallet({ chains }),
```
Comment (DanielSinclair) [inline examples/with-next-custom-button/pages/_app.tsx:46]: Please revert

File: examples/with-next-mint-nft/pages/_app.tsx [hunk 2]
```diff
@@ -56,7 +60,11 @@ const connectors = connectorsForWallets([
   ...wallets,
   {
     groupName: 'Other',
-    wallets: [argentWallet({ chains }), trustWallet({ chains })],
+    wallets: [
```
Comment (DanielSinclair) [inline examples/with-next-mint-nft/pages/_app.tsx:63]: Please revert

File: examples/with-next-siwe-iron-session/pages/_app.tsx [hunk 3]
```diff
@@ -49,6 +50,7 @@ const connectors = connectorsForWallets([
     wallets: [
       argentWallet({ chains }),
       trustWallet({ chains }),
+      okxWallet({ chains }),
```
Comment (DanielSinclair) [inline examples/with-next-siwe-iron-session/pages/_app.tsx:53]: Please revert

File: examples/with-next-siwe-next-auth/pages/_app.tsx [hunk 4]
```diff
@@ -47,6 +48,7 @@ const connectors = connectorsForWallets([
     wallets: [
       argentWallet({ chains }),
       trustWallet({ chains }),
+      okxWallet({ chains }),
```
Comment (DanielSinclair) [inline examples/with-next-siwe-next-auth/pages/_app.tsx:51]: Please revert

File: packages/example/pages/_app.tsx [hunk 5]
```diff
@@ -18,6 +18,7 @@ import {
   argentWallet,
   imTokenWallet,
   ledgerWallet,
+  okxWallet,
```
Comment (DanielSinclair) [inline packages/example/pages/_app.tsx:21]: Please revert

File: site/components/Provider/Provider.tsx [hunk 6]
```diff
@@ -35,6 +36,7 @@ const connectors = connectorsForWallets([
     wallets: [
       argentWallet({ chains }),
       trustWallet({ chains }),
+      okxWallet({ chains }),
```
Comment (DanielSinclair) [inline site/components/Provider/Provider.tsx:39]: Please revert

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Only need OKX added to `packages/example` for testing. Please revert other additions.

File: examples/with-next/pages/_app.tsx [hunk 7]
```diff
@@ -42,6 +43,7 @@ const connectors = connectorsForWallets([
     wallets: [
       argentWallet({ chains }),
       trustWallet({ chains }),
+      okxWallet({ chains }),
```
Comment (DanielSinclair) [inline examples/with-next/pages/_app.tsx:46]: Please revert. For testing, add to `packages/example`

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Comment by DanielSinclair** [comment]
@okxwallet Am seeing linting errors. Run `pnpm run lint:fix` to resolve. The `rainbowkit-example` will deploy upon the next push.

**Comment by okxwallet** [comment]
@DanielSinclair lint erros fixed

File: packages/rainbowkit/src/wallets/walletConnectors/okxWallet/okxWallet.ts [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  return {
+    id: 'okx',
+    name: 'OKX Wallet',
+    iconUrl: async () => (await import('./okxWallet.svg')).default,
+    iconAccent: '#000',
+    iconBackground: '#000',
+    downloadUrls: {
+      browserExtension:
+        'https://chrome.google.com/webstore/detail/okx-wallet/mcohilncbfahbmgdjkbpemcciiolgcge',
+      android: 'https://static.byucvwj.cn/upgradeapp/okx-android.apk',
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/okxWallet/okxWallet.ts:39]: Can we use a Google Play link instead? Or a fallback download landing page? Want to mitigate malicious distributions

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Comment by DanielSinclair** [comment]
@okxwallet Am seeing a few problems on our test dApp [here](https://rainbowkit-example-1ilqr2ju5-rainbowdotme.vercel.app/)
1. Mobile WalletConnect hangs on scan
IMG_5827
2. Injected provider is not detected on desktop:
https://user-images.githubusercontent.com/4412473/217326420-d5829183-80a8-4d8b-8c1e-f939707687a6.mov

**Comment by okxwallet** [comment]
@DanielSinclair 
1. I scan wc qr code, it passed, It seems to be network error?
https://user-images.githubusercontent.com/122091289/217433566-1614d796-597f-4ee5-95d8-3e364304ea8f.mp4
2. Injected provider is not detected on desktop
I tested, it passed.
https://user-images.githubusercontent.com/122091289/217433206-a7f0faff-8e7b-414a-a40e-f0636d8b565b.mov

**Comment by DanielSinclair** [comment]
@okxwallet
1. This is working for me again! ✅
2. Can't seem to get the injector working. A few of the other wallet connectors we are evaluating have run into this issue because they are slow to inject or otherwise executing other work before making `window.ethereum` available to RainbowKit, which detects 'installed' connectors on page load. Could this race condition also affect your wallet extension?

https://user-images.githubusercontent.com/4412473/217435655-b4581d54-e245-4262-876d-f27f3bfe3234.mov

**Comment by okxwallet** [comment]
@DanielSinclair 
Hi, The wallet inject window.ethereum at 'document_start', It defined by extension rules(https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts).
It can work in my chrome in https://rainbowkit-example-git-fork-okxwallet-feat-c75bd8-rainbowdotme.vercel.app.
Please help me check Chrome Developer Tools have errors and Can you give me the version for okx wallet?
I will let more team members test it.

**Comment by DanielSinclair** [comment]
@okxwallet The provider isn't detected and I am unable to connect in this test deployment: https://rainbowkit-example-35f7d4fot-rainbowdotme.vercel.app/

**Comment by okxwallet** [comment]
I test it. The situation you mentioned will indeed happen, not only our wallet, but also the turst wallet. The local environment debugging confirms that there is no problem. Is it possible that it is a problem with the deployment code?

https://user-images.githubusercontent.com/122091289/220924005-a2923e55-4f97-4adf-8bd4-19c3337aacd3.mov

**Comment by DanielSinclair** [comment]
@okxwallet Chrome treats localhost differently. Confirmed to not be an issue with the deployment.

**Comment by okxwallet** [comment]
@DanielSinclair Hi
 The trust wallet have the same problem, Do i write the error code? I follow the document add the code.

https://user-images.githubusercontent.com/122091289/221096265-32000de5-b906-4540-8603-0effaac55d77.mov

File: packages/rainbowkit/src/wallets/walletConnectors/okxWallet/okxWallet.ts [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+      qrCode: 'https://www.okx.com/web3',
+    },
+    createConnector: () => {
+      const connector = shouldUseWalletConnect
+        ? getWalletConnectConnector({ chains })
+        : new InjectedConnector({
+            chains,
+          });
+
+      const getUri = async () => (await connector.getProvider()).connector.uri;
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/okxWallet/okxWallet.ts:45]: On iOS this URI will default to `wc://` and iOS selects a wallet based on the user's usage pattern. Does OKX have a Deep Link or Universal Link for this pairing flow? We will need to be explicit like so:
```
const getUri = async () => {
   const { uri } = (await connector.getProvider()).connector;
   return isAndroid()
     ? uri
     : `https://rnbwapp.com/wc?uri=${encodeURIComponent(uri)}`;
};
```

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Comment by DanielSinclair** [comment]
Closing in favor of #1107

**Pull request closed without merging**